### PR TITLE
Permit 4D image mask in fmri_dataset if time dimension is 1

### PR DESCRIPTION
**Pull request by effigies**
If you create a mask from a PyMVPA dataset, it will get written with as a 4D image with time dimension of 1. This change makes it possible to use those masks without using nibabel to squeeze out that dimension.

### CODE REVIEW

**Comment by nno** [comment]
Thanks for the PR. Could you provide a unit test please?

**Comment by effigies** [comment]
New test passes locally.

File: mvpa2/tests/test_niftidataset.py [hunk 1]
```diff
@@ -442,3 +442,27 @@ def test_strip_nibabel():
...
+    mask4d = pathjoin(pymvpa_dataroot, 'mask4d.nii.gz')
+    # mask4dfail.nii.gz is mask.nii.gz copied twice in the 4th dimension
+    mask4df = pathjoin(pymvpa_dataroot, 'mask4dfail.nii.gz')
+    assert_equal(nibabel.load(mask).shape + (1,),
+                 nibabel.load(mask4d).shape)
+
+    bold1 = fmri_dataset(bold, mask=mask)
+    bold2 = fmri_dataset(bold, mask=mask4d)
+    assert_equal(bold1.shape, bold2.shape)
+    try:
```
Comment (nno) [inline mvpa2/tests/test_niftidataset.py]: How about using ``assert_raises``? here?

**Comment by effigies** [comment]
Fixed.

**Comment by nno** [comment]
Thanks, looks good.

**Pull request merged**